module-system/sov-modules-macros/tests/dispatch/derive_event.rs

```diff
@@ -12,12 +12,35 @@ struct Runtime<C: Context>
 {
     pub first: first_test_module::FirstTestStruct<C>,
     pub second: second_test_module::SecondTestStruct<C>,
+    pub third: third_test_module::ThirdTestStruct<C, u32>,
 }
 
 fn main() {
     // Check to see if the runtime events are getting initialized correctly
-    let _event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum1(10));
-    let _event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum2);
-    let _event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum3(vec![1;3]));
-    let _event = RuntimeEvent::<DefaultContext>::second(second_test_module::Event::SecondModuleEnum);
+    {
+        let event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum1(10));
+        assert_eq!(event.get_key_string(), "first-FirstModuleEnum1");
+    }
+
+    {
+        let event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum2);
+        assert_eq!(event.get_key_string(), "first-FirstModuleEnum2");
+    }
+
+    {
+        let event = RuntimeEvent::<DefaultContext>::first(first_test_module::Event::FirstModuleEnum3(vec![1; 3]));
+        assert_eq!(event.get_key_string(), "first-FirstModuleEnum3");
```

Comment:
To create the key string using the `get_key_string` function, we require a value of `Event::FirstModuleEnum3(..)`. However, obtaining this value necessitates providing a `Vec` this is fine when we save to the db.  

Now, let's consider a scenario where a user wishes to query the database using the `first-FirstModuleEnum3` key. What will be the method for the user to calculate the `first-FirstModuleEnum3` since it doesn't have a value of the `Event::FirstModuleEnum3(...)`? We have to think about the cases where the enums contain some complex structures and possibly other nested enums.